### ISSUE TITLE
Problems following the tutorial

### ISSUE THREAD

**Issue by 420tech** [aspnet/Docs#7291]
I am struggling following the tutorial as written. When I try to add a Razor page using the Entity Framework (CRUD), I get an error that says: "There was an error running the selected code generator. Package restore failed. Rolling back package...."

I know this isn't the support forum, but it is really tough working through the tutorials with no information on why the documented step could not complete.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6719f08e-3bd7-dc1a-71df-f2ef9fbca9d8
* Version Independent ID: 7096fdb3-612e-9e00-bd0b-8ea4886a09ce
* Content: [Add a model to a Razor Pages app in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/model?view=aspnetcore-2.1)
* Content Source: [aspnetcore/tutorials/razor-pages/model.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/tutorials/razor-pages/model.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by d-dizhevsky** [comment]
Could you be more specific and give us more info about the issue? 
What your dev environment setup looks like (target framework, VS version, SDK version, etc etc)?
I'm guessing it is VS/Nuget-related issue with your setup so try to verify that you have proper nuget sources specified.

**Comment by 420tech** [comment]
I followed instructions directly from the tutorial. Running VS 2017 Pro. Entity Framework 6.2 ASP.Net core 2.1. This is a new install on a new Win10 box (first project with VS).

**Comment by Rick-Anderson** [comment]
@420tech  what version of Visual Studio and .Net core SDK?

`dotnet --version`

**Comment by 420tech** [comment]
Visual Studio 15.7.4. ASP.Net core 2.1.301.

**Comment by Rick-Anderson** [comment]
@420tech  are you following the 2.1 version of the tutorial? What step gives you an error?
That's a really old error message we saw in the 2.0 version.

**Comment by 420tech** [comment]
The second step - "Add a model". When I tried to generate a razor page using the "Add Scaffold.." and Razor Pages using Entity Framework (CRUD).

**Comment by d-dizhevsky** [comment]
@420tech  I assume you are using this sample? https://github.com/aspnet/Docs/tree/master/aspnetcore/tutorials/razor-pages/razor-pages-start/sample/RazorPagesMovie

**Comment by 420tech** [comment]
I am writing the code myself directly from the tutorial. I am not using any of the code from GitHub. The tutorial I am following can be found here: https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/model?view=aspnetcore-2.1

The step it fails on is the "Add a model". The Add Scaffold part fails on the code generation. Everything I can see tells me I am using 2.1 (e.g. in the Dependencies under NuGet is EntityFramework 6.2, Microsoft.AspNetCore 2.1 and related aspnetcore - all 2.1).

**Comment by d-dizhevsky** [comment]
That's odd, could you push your project to the repo so we can clone it and try to repro this on our own?

**Comment by Rick-Anderson** [comment]
@420tech  try this

```
dotnet new webapp -o RazorPagesMovie
cd RazorPagesMovie
dotnet add package Microsoft.VisualStudio.Web.CodeGeneration.Design

dotnet tool install -g dotnet-aspnet-codegenerator --version 2.1.0

// Add movie model to project
// Build run the app.
//  From project directory, run in a command window

dotnet aspnet-codegenerator razorpage -m Movie -dc RazorPagesMovie.Data.MovieContext -udl -outDir Pages\Movies --referenceScriptLibraries

dotnet ef migrations add Initial
dotnet ef database update
```

**Comment by 420tech** [comment]
Thanks @Rick-Anderson! That worked!

**Comment by Berkyle** [comment]
I'm not sure if anyone else is having the same issue anymore but a coworker fixed this error for me by building the web application and THEN creating a new razor page.

**Comment by GavinGuoweiXu** [comment]
I have fixed it, by install the package Microsoft.VisualStudio.Web.CodeGeneration.Design from nuget.

**Comment by haybatov** [comment]
In my case, the problem was caused by missing Microsoft.EntityFrameworkCore.Tools NuGet package. Installing it solved the problem.

**Comment by NoorAlazawi** [comment]
@Rick-Anderson Thank u sooo much it solve my problem

**Comment by hamedsouri** [comment]
Just connect to the Internet to download and install the package.

**Comment by DiverBW** [comment]
A year later, it seems this tutorial has not been updated. I am new to .Net Core, and am also following this tutorial:

I started on this page:
https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/razor-pages-start?view=aspnetcore-5.0&tabs=visual-studio

I then clicked on Next: Add a Model which took me to this page:

https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/model?view=aspnetcore-5.0&tabs=visual-studio

When I get to the section where I'm adding a new scaffolded item (CRUD), it fails. It seems it is trying to automatically add Entity Framework related NuGet packages that are only supported on .Net Standard, and this is a .Net Core 5 application.

Is there a new tutorial I should be following?

**Comment by Rick-Anderson** [comment]
> ckages that are only supported on .Net Standard, and this is a .Net Core 5 application.

There is no such thing as .NET Core 5. This tutorial works with .NET 5.  I'm not sure what you're doing wrong, but it works.

**Comment by DiverBW** [comment]
Yes, you are correct. I misspoke. The target framework is set to .Net 5.0.

The error seems to have something to do with the scaffolding sequence trying to do some sort of background add of additional NuGet packages, but I see no diagnostic error output other than this error dialog.

I added the following NuGet packages via the NuGet  package manager:

- Microsoft.EntityFrameworkCore
- Microsoft.EntityFrameworkCore.Abstractions
- Microsoft.EntityFrameworkCore.Analyzers
- Microsoft.EntityFrameworkCore.Design
- Microsoft.EntityFrameworkCore.Relational
- Microsoft.EntityFrameworkCore.Tools

I then tried to add the scaffolding again. This time I saw a quick dialog that indicates it was trying to add the NuGet package Microsoft.EntityFrameworkCore.SqlServer.  But that transient dialog quickly went away and the same error dialog appeared.

I tried to add Microsoft.EntityFrameworkCore.SqlServer, but it failed due to lots of things not being compatible with .Net 5.0. Studying its dependencies, it looked like the only thing it needed that I had not already referenced was Microsoft.Data.SqlClient, so I tried adding that. Same thing, it had dependencies to old libraries that were not compatible with .Net 5.0.

Although I am following the tutorial, I had tried jumping ahead a bit by creating my own model class.  I am going to create a new project and go through the tutorial again, this time following it exactly (i.e. creating the Movie class) and see if that works. Perhaps that will give me a hint as to where I went off track.

![2021-08-24_10-16-56](https://user-images.githubusercontent.com/13752261/130634318-a4b5156d-586a-4923-8abb-c84782cb2484.png)

**Comment by DiverBW** [comment]
Found this suggestion: go to Tools -> Options -> Nuget Package Manager ->Package Manager Setting-> General -> Clear All Nuget Cache(s). 

Once I did that, the dialogs during the scaffolding wizard run were a bit more informative and clearly told me it was trying to add more NuGet packages.

But eventually it still failed.

But somewhere in this sequence it started flashing a warning dialog saying I was going to overwrite the Index.cshtml file. I had been doing the right click to start the scaffolding run on the top "Pages" folder, which did have an Index.cshtml file.  So for grins, I created a new subfolder under the Pages folder, then ran the scaffolding wizard in that folder. It worked!

So this problem, from my experience, seems to have something to do with the fact that the scaffolding wizard generates an Index.cshtml file and there better not be one already there!

Bad diagnostics from Visual Studio for sure, but hopefully my post will help someone else get past this step.

**Comment by francisprojects33** [comment]
I think the fix is to update your Visual Studio to the latest using **Check for update** in Help menu.

**Comment by casswoo** [comment]
> Yes, you are correct. I misspoke. The target framework is set to .Net 5.0.
> 
> The error seems to have something to do with the scaffolding sequence trying to do some sort of background add of additional NuGet packages, but I see no diagnostic error output other than this error dialog.
> 
> I added the following NuGet packages via the NuGet package manager:
> 
> * Microsoft.EntityFrameworkCore
> * Microsoft.EntityFrameworkCore.Abstractions
> * Microsoft.EntityFrameworkCore.Analyzers
> * Microsoft.EntityFrameworkCore.Design
> * Microsoft.EntityFrameworkCore.Relational
> * Microsoft.EntityFrameworkCore.Tools
> 
> I then tried to add the scaffolding again. This time I saw a quick dialog that indicates it was trying to add the NuGet package Microsoft.EntityFrameworkCore.SqlServer. But that transient dialog quickly went away and the same error dialog appeared.
> 
> I tried to add Microsoft.EntityFrameworkCore.SqlServer, but it failed due to lots of things not being compatible with .Net 5.0. Studying its dependencies, it looked like the only thing it needed that I had not already referenced was Microsoft.Data.SqlClient, so I tried adding that. Same thing, it had dependencies to old libraries that were not compatible with .Net 5.0.
> 
> Although I am following the tutorial, I had tried jumping ahead a bit by creating my own model class. I am going to create a new project and go through the tutorial again, this time following it exactly (i.e. creating the Movie class) and see if that works. Perhaps that will give me a hint as to where I went off track.
> 
> ![2021-08-24_10-16-56](https://user-images.githubusercontent.com/13752261/130634318-a4b5156d-586a-4923-8abb-c84782cb2484.png)

I solved the issue by rolling back to v5.0.9 from v5.0.10!

![image](https://user-images.githubusercontent.com/8214410/134259716-565f2e9c-62aa-490e-ae7d-8d37e1e4c5e4.png)

**Comment by GaryITSupport** [comment]
i had this problem, it was because i only had the visual studio offline nuget source, adding the nuget.org as a nuget source fixed it. see here https://stackoverflow.com/questions/69045231/nuget-package-sources-missing
